sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1788,6 +1788,15 @@ val SHUFFLE_COMPRESSION_LZ4_CHUNK_SIZE = conf("spark.rapids.shuffle.compression.
         .integerConf
         .createWithDefault(20)
 
+  val SHUFFLE_GPU_SERDE_ENABLED =
+    conf("spark.rapids.shuffle.serde.enabled")
```

Comment:
Lets change this to:
```
spark.rapids.shuffle.serde.type
```

And there are two types so far: "CPU", "GPU". The way the flag is used is fine we would still convert it to a boolean `isGpuSerdeEnabled`, but we would test whether `spark.rapids.shuffle.serde.type=="GPU"`.

Reply:
It's internal right now, but as we learn more about this method we need to add documentation that says when to use which. Or come up with smart heuristics that will pick CPU/GPU automatically (so we can add another type.. AUTO)

Reply:
Done